src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1457,10 +1457,23 @@ private static bool IsConditional(IBinaryOperation operation)
         public override IOperation VisitBinaryOperator(IBinaryOperation operation, int? captureIdForResult)
         {
             if (IsConditional(operation) &&
-                operation.OperatorMethod == null && // PROTOTYPE(dataflow): cannot properly handle user-defined conditional operators yet.
-                operation.Type.SpecialType == SpecialType.System_Boolean) // PROTOTYPE(dataflow): cannot properly handle nullable conditional operators yet.
+                operation.OperatorMethod == null) // PROTOTYPE(dataflow): cannot properly handle user-defined conditional operators yet.
             {
-                return VisitBinaryConditionalOperator(operation, sense: true, captureIdForResult, fallToTrueOpt: null, fallToFalseOpt: null);
+                if (ITypeSymbolHelpers.IsBooleanType(operation.Type) &&
+                    ITypeSymbolHelpers.IsBooleanType(operation.LeftOperand.Type) &&
+                    ITypeSymbolHelpers.IsBooleanType(operation.RightOperand.Type))
```

Comment:
Can any of these types be null for error case? For example, if one of the operands is `null` literal.

Reply:
> Can any of these types be null for error case? For example, if one of the operands is null literal.

I do not think it is possible given the current implementation of compilers.

---
In reply to: [189607822](https://github.com/dotnet/roslyn/pull/26994#discussion_r189607822) [](ancestors = 189607822)